### DISCUSSION TITLE
Replication Problem - External replica cluster does not create secrets

### DISCUSSION THREAD

**Discussion by eyanez111** [cloudnative-pg/cloudnative-pg#5540] [Q&A]
Hello everyone,
The functionality that an external replica cluster does not create an ***-app secret (basic-auth secret) until it is promoted to primary? I'm trying to find out why this is the case and if there is a configuration to ensure that secret is always available, even on the replica cluster. The secret being non-existent breaks my services. Secondly, when promoting the replica to primary, the secret does get created, but the authentication fails anyways.

we are using:
kuberntes 1.26 rancher on both clusters
Postgress 16.4
CNPG 1.24.0
Please if someone can point me in the direction of some documentation supporting the functionality OR point me to a channel that may help resolve this. 

Thank you.

**Comment by boris-savic** [comment]
+1 same question
